pkg/sendbuffer/sendbuffer.go

```diff
@@ -170,34 +172,46 @@ func (sb *SendBuffer) sendAndPurge() error {
 	}
 
 	if err := sb.sender.Send(toSendBuff); err != nil {
-		sb.logger.Log("msg", "failed to send, dropping data", "err", err)
+		sb.logger.Log("msg", "failed to send, will retry", "err", err)
+		return nil
 	}
+	// TODO: populate logs with device data (id, serial, munemo, orgid) when we
+	// get first set of control data with device info before shipping
+
+	// testing on a new enrollment in debug mode, log size hit 130K bytes
+	// before enrollment completed and was able to ship logs
+	// 2023-11-16
+	sb.writeMutex.Lock()
+	defer sb.writeMutex.Unlock()
+	sb.deleteLogs(lastKey)
```

Comment:
Should these mutexes be in deleteLogs?

Reply:
the write method may also call deleteLogs if at capacity it also locks the mutex